client/src/lib/content/hooks/useGetExercisesByTags.tsx

```diff
@@ -0,0 +1,28 @@
+import {intersection, take} from 'ramda';
+import {useMemo} from 'react';
+import useAppState from '../../appState/state/state';
+import useExercises from './useExercises';
+import {Tag} from '../../../../../shared/src/types/generated/Tag';
+
+const useGetExercisesByTags = (tags: Tag[], excludeId?: string) => {
+  const {showHiddenContent} = useAppState(state => state.settings);
+  const exercises = useExercises();
+  const tagNames = useMemo(() => tags?.map(t => t.tag), [tags]);
+
+  return useMemo(() => {
+    if (!tagNames) return [];
+    const exerciseList = exercises
+      .filter(e => showHiddenContent || !e.hidden)
+      .filter(
+        e =>
+          intersection(
+            (e.tags || []).map(t => t.tag),
+            tagNames,
+          ).length > 0 && e.id !== excludeId,
+      );
+
+    return take(5, exerciseList);
```

Comment:
Can we remove the limiting from here and do it outside of the hook? Or make it an argument. We will be using this hook when filtering all exercises in explore soon.

Reply:
Another approach could be to split this into one pure `useGetExercisesByTags` that doesn't exclude or limit. And then a `useGetRelatedExercises` that uses `useGetExercisesByTags` but also excludes and limits the result.